webapp/app/[locale]/stake/_components/tokenRewards.tsx

```diff
@@ -0,0 +1,26 @@
+import { ReactNode } from 'react'
+
+import { HemiTag } from './rewardTag'
+
+interface Props {
+  rewards: string[]
+}
+
+// TODO: It's pending to implement some missing tags
+// Issue #799 https://github.com/hemilabs/ui-monorepo/issues/799
+const rewardComponentMap: Record<string, ReactNode> = {
+  hemi: <HemiTag />,
+}
+
+export function TokenRewards({ rewards }: Props) {
+  if (!rewards || rewards.length === 0) {
+    return null
+  }
+  return (
+    <>
+      {rewards.map(reward => (
+        <div key={reward}>{rewardComponentMap[reward] || null}</div>
```

Comment:
If we do the updates mentioned of typings `rewards` (so it is not `string[]`), you can ensure this will always be defined, so we can remove `|| null`

```suggestion
        <div key={reward}>{rewardComponentMap[reward]}</div>
```

Nevertheless, if it wasn't defined, `rewardComponentMap[reward]` returns `undefined` which renders nothing, so it could be skipped as well.

Going even beyond that, I think if you add the key for each component where there are defined in line 12, like 

```tsx
const rewardComponentMap: Record<StakeToken['extensions']['rewards'], ReactNode> = {
  hemi: <HemiTag key="hemi" />,
}
```

you can remove the extra div, like

```tsx
<>
  {rewards.map(reward => rewardComponentMap[reward])}
</>
```

if you don't feel like defining the key above, you can use fragments to also skip the extra `div`

```tsx
<>
      {rewards.map(reward => <React.Fragment key={reward}>{rewardComponentMap[reward]}</React.Fragment>)
</>
```

Reply:
Done. Thanks